### PR TITLE
About / What We Do

### PR DESCRIPTION
**Pull request by vuonga1103**
### Ticket
[https://trello.com/c/4CcBJZgY/69-what-we-do](https://trello.com/c/4CcBJZgY/69-what-we-do)

### Checklist
- [ ] There are no warnings in the browser console _(No new warnings, but persisting warnings regarding deprecation of SharedArrayBuffer and axes issues remained)_
- [X] I made sure the component is WCAG compliant using ANDI
- [X] I made sure the UI follows the style guide in the wireframe (font, font-size, etc...)
- [X] I made sure that the screen reader can read all the text
- [X] I made sure I removed all the console.logs
- [X] I made sure mobile portrait view looks good
- [X] I made sure mobile landscape view looks good
- [X] I added storybook for the component
	- [ ] Storybook not applicable
- [X] I added unit test coverage
	- [ ] I added integration test coverage
	- [ ] Unit test coverage not applicable

### To Do
- [ ] Need to replace cat photos in "Who We Are" section with team member photos from API

### Screenshot
Screen Shot 2021-04-14 at 5 16 49 PM
Screen Shot 2021-04-14 at 5 16 31 PM

### CODE REVIEW

File: src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        break;
+      default:
+        <WhatWeDo />;
+    }
+    return result;
+  })();
+
+  return (
+    <section className={container}>
+      <article className={content}>{page}</article>
```
Comment (cfernando415) [inline src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js:29]: invoke the func

`<article className={content}>{page()}</article>`

**Review by cfernando415 — CHANGES_REQUESTED** [review]
please see comment

File: src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js (hunk 1 above)
Comment (vuonga1103) [reply src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js:29]: `page` is an IIFE so it is already invoked. Do you think IIFE is ok in this case?

Comment (vuonga1103) [reply src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js:29]: or should I stick to regular function to avoid confusion

Comment (cfernando415) [reply src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js:29]: personally, not really a fan of IIFE, unless there is a need to (just my stylistic preference i guess?); for example, you're other scenario of IIFE in the WhoWeAre component is acceptable cause you're passing it as a props and using the object's name as the prop's name (line 57 and 58)...
image

File: src/components/AboutContainer/AboutContainer.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    });
+  });
+
+  it('should display corresponding title when an option is clicked', async () => {
+    await optionLabels.forEach((optionLabel) => {
+      let optionLink = screen.getByTestId(optionLabel);
+      let title = screen.getByTestId('title');
+      fireEvent.click(optionLink);
+      expect(title).toHaveTextContent(optionLabel);
+    });
```
[the remark this answers is not in the document]
Comment (vuonga1103) [reply src/components/AboutContainer/AboutContainer.test.js]: Fixed in 1944187

File: src/components/AboutContainer/AboutContainer.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+describe('<AboutContainer />', () => {
+  beforeEach(() => {
+    act(() => {
+      render(
+        <Router>
+          <AboutContainer />
+        </Router>
+      );
+    });
```
[the remark this answers is not in the document]
Comment (vuonga1103) [reply src/components/AboutContainer/AboutContainer.test.js]: Fixed in 1944187

File: src/components/AboutContainer/AboutContainer.test.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    expect(image.src).toContain('two-women');
+  });
+
+  it('should display navigation bar with correct options', () => {
+    optionLabels.forEach((optionLabel) => screen.getByTestId(optionLabel));
+  });
+
+  it('should fire a click event for all menu options', () => {
+    optionLabels.forEach((optionLabel) => {
+      let optionLink = screen.getByTestId(optionLabel);
```
[the remark this answers is not in the document]
Comment (vuonga1103) [reply src/components/AboutContainer/AboutContainer.test.js]: Fixed in 1944187

File: src/components/AboutContainer/AboutContentContainer/AboutContentPages/Programs.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { Link, Typography } from '@material-ui/core';
+import useStyles from '../AboutContentContainer.styles';
+
+const Programs = () => {
+  const cohort = 'Spring 2021';
+  const appDeadline = 'January 1st 2021';
```
[the remark this answers is not in the document]
Comment (vuonga1103) [reply src/components/AboutContainer/AboutContentContainer/AboutContentPages/Programs.js]: Fixed in c13fd83f924bae231e92239aab0f5b3dc1948020

File: src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js (hunk 1 above)
Comment (vuonga1103) [reply src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js]: Fixed in 8fec142ba0870f95295740ed59241bb935ef2872

File: src/components/AboutContainer/AboutSideBar/AboutSideBar.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import useStyles from './AboutSideBar.styles';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
```
Comment (PeaWarrior) [inline src/components/AboutContainer/AboutSideBar/AboutSideBar.js:3]: Nothing super constructive. Just prefer to use their icon API
`import { ChevronRight, ChevronLeft } from '@material-ui/icons';`

File: src/components/AboutContainer/AboutSideBar/AboutSideBar.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+import { Link } from '@material-ui/core';
+import { useHistory } from 'react-router-dom';
+import { PropTypes } from 'prop-types';
+import { useMediaQuery } from '@material-ui/core';
```
Comment (PeaWarrior) [inline src/components/AboutContainer/AboutSideBar/AboutSideBar.js:7]: Same with line 4 and 7. Can be combined

**Review by PeaWarrior — CHANGES_REQUESTED** [review]
Also I found a weird CSS issue when screen width is at 799px.
Reproducible in Chrome and Safari.
Screen Shot 2021-04-15 at 1 22 34 AM
Screen Shot 2021-04-15 at 1 22 56 AM

File: src/components/AboutContainer/AboutSideBar/AboutSideBar.js (hunk 6 above)
Comment (vuonga1103) [reply src/components/AboutContainer/AboutSideBar/AboutSideBar.js]: Fixed in b57f2c7f4a31b6e2bfb46523d020db6754cdb21e

File: src/components/AboutContainer/AboutSideBar/AboutSideBar.js (hunk 7 above)
Comment (vuonga1103) [reply src/components/AboutContainer/AboutSideBar/AboutSideBar.js]: Fixed in b57f2c7f4a31b6e2bfb46523d020db6754cdb21e

**Comment by vuonga1103** [comment]
weird CSS issue when screen width is at 799px is fixed

File: src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        result = <WhoWeAre />;
+        break;
+      case 'Programs':
+        result = <Programs />;
+        break;
+      default:
+        <WhatWeDo />;
+    }
+    return result;
+  };
```
Comment (cfernando415) [inline src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js:25]: let's rewrite this.. get rid of the variable result and the break statement... return the component

**Review by cfernando415 — CHANGES_REQUESTED** [review]
pls see comments

File: src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js (hunk 8 above)
Comment (vuonga1103) [reply src/components/AboutContainer/AboutContentContainer/AboutContentContainer.js]: Good idea, refactored in b261349176882a861ea91d2a5d4ec317ff829e91

**Comment by cfernando415** [comment]
merge conflict

**Review by cfernando415 — APPROVED** [review]
:+1:

**Review by PeaWarrior — APPROVED** [review]
👍

**Pull request merged**